maps/_map.dm

```diff
@@ -146,6 +146,18 @@ proc/get_base_turf(var/z)
 	var/datum/zLevel/L = map.zLevels[z]
 	return L.base_turf
 
+proc/change_base_turf(var/choice,var/new_base_path,var/update_old_base = 0)
+	if(update_old_base)
+		for(var/turf/T in turfs)

```

Comment:
Byond is bad and I profiled it to be cheaper with the current method by a considerable margin 
![profile](https://cloud.githubusercontent.com/assets/7193289/11992666/81dcb24c-aa20-11e5-8eae-a09738197a0c.png)